padertorch/contrib/examples/source_separation/pit/train.py

```diff
@@ -79,104 +81,141 @@ def config():
-    print(f"make train")
-    print()
-    print('or')
-    print()
-    print(f"cd {_config['trainer']['storage_dir']}")
-    print('make ccsalloc')
 
 
-@ex.capture
-def prepare_and_train(_config, _run, train_dataset, validate_dataset, database_json):
-    """ Prepares the train and validation dataset from the database object """
+def prepare(_config):
+    """
+    Preparation of the train and validation datasets for the training and initialisation of the padertorch trainer,
+    using the configuration dict.
+    Args:
+        _config: Configuration dict of the experiment
+
+    Returns:
+    3-Tuple of the prepared datasets and the trainer.
+        trainer: padertorch trainer
+        train_dataset: training_dataset
+        validate_dataset: dataset for validation
+    """
+
+    # Extraction needed strings from the config dict
+    train_dataset_name = _config['train_dataset_name']
+    validate_dataset_name = _config['validate_dataset_name']
+    database_json = _config['database_json']
 
-    sacred.commands.print_config(_run)
+    # Initialisation of the trainer
     trainer = pt.Trainer.from_config(_config["trainer"])
-    checkpoint_path = trainer.checkpoint_dir / 'ckpt_latest.pth'
-
     db = JsonDatabase(json_path=database_json)
-    print(repr(train_dataset), repr(validate_dataset))
 
-    trainer.test_run(
-        prepare_dataset_captured(db, train_dataset),
-        prepare_dataset_captured(db, validate_dataset),
-    )
-    trainer.register_validation_hook(
-        prepare_dataset_captured(db, validate_dataset)
-    )
-    trainer.train(
-        prepare_dataset_captured(db, train_dataset),
-        resume=checkpoint_path.is_file()
-    )
+    # Preparation of the datasets
+    train_dataset = prepare_dataset(db, train_dataset_name,
+                                    _config['batch_size'],
+                                    prefetch = not _config['debug'])
+
+    validate_dataset = prepare_dataset(db, validate_dataset_name,
+                                       _config['batch_size'],
+                                       prefetch = not _config['debug'])
+
+    # Print the representations of the two datasets to the console.
+    print(repr(train_dataset_name), repr(validate_dataset_name))
+
+    return (trainer, train_dataset, validate_dataset)
+
 
+def train(_config, trainer, train_dataset, validate_dataset):
+    """
+    Starts the training, with the passed trainer and the passed datasets.
+    Creates checkpoints of the training, according to the trainer configuration.
+
+    Args:
+        _config: Configuration dict of the experiment
+        trainer: padertorch trainer
+        train_dataset: training_dataset
+        validate_dataset: dataset for validation
+
+    Returns:
+        None
+    """
+
+    # Test run to detects possible errors in the trainer/datasets
+    trainer.test_run(train_dataset, validate_dataset)
+
+    # path where the checkpoints of the training are stored
```

Comment:
This comment is lower case, others are upper case. Stick to one (I prefer upper case)